tensorflow/lite/micro/micro_allocator.cc

```diff
@@ -221,21 +221,24 @@ TfLiteStatus AllocationInfoBuilder::GetOfflinePlannedOffsets(
 
 TfLiteStatus AllocationInfoBuilder::AddScratchBuffers(
     internal::ScratchBufferRequest* scratch_buffer_requests,
-    ScratchBufferHandle* scratch_buffer_handles) {
+    ScratchBufferHandle* scratch_buffer_handles, int subgraph_idx) {
   // Set up allocation info for buffers.
-  for (size_t i = tensor_count_; i < tensor_count_ + buffer_count_; ++i) {
+  for (size_t i = 0, buffers_added = 0; i < buffer_count_; ++i) {
     internal::ScratchBufferRequest* current_request =
-        &(scratch_buffer_requests[i - tensor_count_]);
-    ScratchBufferHandle* current_handle =
-        &(scratch_buffer_handles[i - tensor_count_]);
-
-    AllocationInfo* current = &info_[i];
-    current->output_ptr = reinterpret_cast<void**>(&current_handle->data);
-    current->bytes = current_request->bytes;
-    current->first_created = current_request->node_idx;
-    current->last_used = current_request->node_idx;
-    current->offline_offset = kOnlinePlannedBuffer;
-    current->needs_allocating = true;
+        &(scratch_buffer_requests[i]);
+    if (current_request->subgraph_idx == subgraph_idx) {
+      ScratchBufferHandle* current_handle =
+          &(scratch_buffer_handles[buffers_added]);
+
+      AllocationInfo* current = &info_[tensor_count_ + buffers_added];
```

Comment:
Might be worthy of adding a comment about why  +tensor_count_?

Reply:
Done.